packages/11ty/_includes/components/WebC.md

```diff
@@ -0,0 +1,241 @@
+**`type="text/javascript"`** blocks are regular client-side `<script>` tags.
+- they need to be `webc:keep`'d if WebC's bundler is not being used
+- setup event listeners and client-side functionality
+
+## Using a WebC component
+The most straightforward way to render a WebC component is in a WebC layout:
+**`layout.webc`**
+```html
+---
+layout: base
+---
+<section class="my-component-section">
+  <my-component gradoo="Gradoo" />
+</section>
+```
+**`my-component.webc`**
+```html
+<div class="my-component" @text="gradoo"></div>
+```
+
+Alternatively, you can use `renderTemplate()` when calling it from a different layout context:
+
+**`layout.liquid`**
+```html
+---
+layout: base.liquid
+---
+{% renderTemplate "webc" %}
+<my-component gradoo="Gradoo"></my-component>
+{% endrenderTemplate %}
+```
+
+**`layout.11ty.js`**
+```js
+module.exports = {
+  data: {
+    layout: "base.11ty.js",
+  },
+  async render() {
+    return await this.renderTemplate(
+      '<my-component gradoo="Gradoo"></my-component',
+      'webc'
+    );
+  }
+}
+```
+
+## Bundling styles + scripts
+See https://www.11ty.dev/docs/languages/webc/#css-and-js-(bundler-mode)
+**`my-component.webc`**
+```html
+<div class="my-component" @text="gradoo"></div>
+<style webc:keep>
+.my-component {
+  background: lemonchiffon;
+}
+</style>
+```
+Like `<script>` tags, `<style>` tags need `webc:keep` to add them to the DOM. To avoid this, add the bundled assets to a layout:
+**`base-layout.webc`**
+```html
+<!doctype html>
+<html lang="$data.publication.language">
+  <head>
+    <meta charset="UTF-8"/>
+    <title @text="$data.publication.title"></title>
+    <style webc:keep @raw="getBundle('css')"></style>
+    <script webc:keep @raw="getBundle('js')"></script>
+  </head>
+  <body>
+    <main @raw="content"></main>
+  </body>
+</html>
+```
+**`my-component.webc`**
+```html
+<div class="my-component" @text="gradoo"></div>
+<style>
+/* these styles are now part of the bundle */
+.my-component {
+  background: lemonchiffon;
+}
+</style>
+```
+
+## Properties + dynamic attributes
+See https://www.11ty.dev/docs/languages/webc/#dynamic-attributes-and-properties
+Regular text attributes work as expected:
+**`my-component.webc`**
+```html
+<div class="my-component" @text="gradoo"></div>
+```
+**`layout.webc`**
+```html
+---
+layout: base
+---
+<section class="my-component-section">
+  <my-component gradoo="Gradoo" />
+</section>
+```
+rendered DOM:
+```html
+<section class="my-component-section">
+  <my-component gradoo="Gradoo">
+    <div class="my-component">Gradoo</div>
+  </my-component>
+</section>
+```
+
+Dynamic attributes are prefixed with a `:`.
+**`layout.webc`**
+```html
+---
+layout: base
+---
+<section class="my-component-section">
+  <my-component :gradoo="gradoo" />
+</section>
+<script webc:setup>
+const gradoo = "Gradoo!!"
+</script>
+```
+rendered DOM:
+```html
+<section class="my-component-section">
+  <my-component gradoo="Gradoo">
+    <div class="my-component">Gradoo</div>
```

Comment:
```suggestion
  <my-component gradoo="Gradoo!!">
    <div class="my-component">Gradoo!!</div>
```
Does the value of `const gradoo` in `<script webc:setup>` get rendered by the component?

Reply:
yep!